DiagnosticsExtension/Controllers/ConnectionStringValidationController.cs

```diff
@@ -72,25 +75,24 @@ public async Task<HttpResponseMessage> Validate([FromBody] ConnectionStringReque
             {
                 return Request.CreateErrorResponse(HttpStatusCode.BadRequest, "Type is not specified in the request body");
             }
-
             var result = await Validate(requestBody.ConnectionString, requestBody.Type);
             return result;
         }
 
         [HttpGet]
         [Route("validateappsetting")]
-        public async Task<HttpResponseMessage> ValidateAppSetting(string appSettingName, string type)
+        public async Task<HttpResponseMessage> ValidateAppSetting(string appSettingName, string type, string entityName = null)
         {
-            var envDict = Environment.GetEnvironmentVariables();
-            if (envDict.Contains(appSettingName))
+            bool success = Enum.TryParse(type, out ConnectionStringType csType);
+            if (success && typeValidatorMap.ContainsKey(csType))
             {
-                var connectionString = (string)envDict[appSettingName];
-                return await Validate(connectionString, type);
+                var result = await typeValidatorMap[csType].ValidateViaAppsettingAsync(appSettingName, entityName);
+                return Request.CreateResponse(HttpStatusCode.OK, result);
             }
             else
             {
-                return Request.CreateErrorResponse(HttpStatusCode.NotFound, $"AppSetting {appSettingName} not found");
+                return Request.CreateErrorResponse(HttpStatusCode.BadRequest, $"Type '{type}' is not supported");
```

Comment:
>return Request.CreateErrorResponse(HttpStatusCode.BadRequest, $"Type '{type}' is not supported"); [](http://example.com/codeflow?start=16&length=97)

We will no longer get a 404 error for not found appsetting, is this change by design?

Reply:
This is intentional as the specified app setting in the case of managed identity based connections is prefix that the appropriate validators will use to load the managed identity configuration.  If neither the connection string (specified app setting) nor managed identity configuration (specified here as the prefix to one or more app settings) are found, an error is returned to the user.  See Constants.cs:29 BlobServiceUriMissingSummary for an example.